my_enumerable_methods.rb

```diff
@@ -0,0 +1,138 @@
+module Enumerable
+  def my_each
+    if block_given?
+      each do |counter|
+        yield(counter)
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0...size).each do |counter|
+        yield self[counter], counter
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    select_in = []
+    my_each do |counter|
+      select_in << yield(counter)
+    end
+    select_in
+  end
+
+  def my_all?
+    my_each do |element|
+      if block_given?
+        yield(element)
+      elsif element.nil? or !i or !yield(element)
+        puts false
+      end
+    end
+    puts true
+  end
+
+  def my_any?
+    my_each do |element|
+      if block_given?
+        yield element
+      elsif element.nil? or element or yield(element)
+        puts true
+      end
+    end
+    puts false
+  end
+
+  def my_none?
+    puts true if block_given?
+    my_each do |element|
+      puts false if yield element
+    end
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_none?(String) == true #true
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words = %w[dog door rod blade]
words.my_none?(5) == words.none?(5) #true
```
- [ ] Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
 words.my_none?(/z/) == words.none?(/z/) #true
 words.my_none?(/d/) == words.none?(/d/) #true
```
- [ ] returns true if the block never returns true for all elements
```ruby
true_array = [nil, false, true, []]
false_array = [nil, false, nil, false]
false_array.my_none? #true
true_array.my_none? #false
```
- [ ] when no block or argument is given
```ruby
true_array = [nil, false, true, []] 
false_array = [nil, false, nil, false]
false_array.my_none? == true #true
true_array.my_none? == false #false
```